### PR TITLE
[feature] Add information about remaining quests in chain in expanded map tooltips

### PR DESCRIPTION
**Pull request by jenswaage**
## Issue references

 Related to #4266 

## Proposed changes
- List remaining quests in a chain in the expanded map tooltip (shift key) w/ option to toggle
- Minor refactors and comments

## Screenshots

Remaining quests in chain for a single quest
single_quest

Remaining quests in chains for multiple quests
multiple quests

New option added
Screenshot 2024-04-01 at 00 29 02

### CODE REVIEW

**Comment by Cabro** [comment]
Looks very good, however i know for sure the DB doesn't have the correct next in chain quest id for even half the quests :-s

**Comment by jenswaage** [comment]
@Cabro isn’t that a separate issue? Or do you think it reduces the usefulness of the feature to the extent that it must be solved first?

**Comment by MyTechnoHunter** [comment]
Does it honor the show questID option like the shift tooltips do?

**Comment by MyTechnoHunter** [comment]
Nice, will help in reporting i would think

**Comment by Cabro** [comment]
> @MyTechnoHunter added support Screenshot 2024-04-06 at 17 48 33

How are the next quests in the chain available if you are a few steps before those?

**Comment by MyTechnoHunter** [comment]
> 
> How are the next quests in the chain available if you are a few steps before those? I'd put a different word instead of "Available" or just remove that completely.
> 
> Another thing, there are many quest chains where you can only get to the next quests if you gain a level, or several. How is it handling that case?

Beyond scope i would think..
Requirements of the chained quests can be handled by lookup, the fact the current quest _is_ a chain quest is more than helpful

**Comment by jenswaage** [comment]
@Cabro 
> How are the next quests in the chain available if you are a few steps before those? I'd put a different word instead of "Available" or just remove that completely.

I agree with this. Changed it to only show when the quest is a group or dungeon quest.
Screenshot 2024-04-06 at 20 41 06
Screenshot 2024-04-06 at 20 19 05

> Another thing, there are many quest chains where you can only get to the next quests if you gain a level, or several. How is it handling that case?

This feature just naively lists all remaining quests that are part of a direct series regardless of level or other requirements. The intention of the feature is just to have a quick way to see if a quest is part of a chain and how long that chain is. I agree with @MyTechnoHunter that any additional information is better handled via some sort of lookup so as to not overload the tooltip with information.

File: Modules/Tooltips/MapIconTooltip.lua [hunk 1]
```diff
@@ -441,28 +493,35 @@ function _MapIconTooltip:IsMinimapInside()
...
 end
 
-function _MapIconTooltip:GetAvailableOrCompleteTooltip(icon)
-    local tip = {};
-    if icon.data.Type == "complete" then
-        tip.type = "(" .. l10n("Complete") .. ")";
+---@param quest Quest
+---@return string tag
+--- Get the quest tag to display in the tooltip
+function GetQuestTag(quest)
```
Comment (BreakBB) [inline Modules/Tooltips/MapIconTooltip.lua:499]:
```suggestion
--- Get the quest tag to display in the tooltip
---@param quest Quest
---@return string tag
local function _GetQuestTag(quest)
```

File: Modules/Tooltips/MapIconTooltip.lua [hunk 2]
```diff
@@ -441,28 +493,35 @@ function _MapIconTooltip:IsMinimapInside()
...
         else
-            tip.type = "(" .. l10n("Available") .. ")";
+            return "(" .. l10n("Available") .. ")";
         end
     end
+end
+
+function _MapIconTooltip:GetAvailableOrCompleteTooltip(icon)
+    local tip = {};
+    tip.type = GetQuestTag(icon.data)
```
Comment (BreakBB) [inline Modules/Tooltips/MapIconTooltip.lua:524]:
```suggestion
    tip.type = _GetQuestTag(icon.data)
```

**Review by BreakBB — APPROVED** [review]
Thank you very much for this @jenswaage and sorry for not looking into this earlier.

I liked that you extracted `_GetQuestTag` 👍🏻

File: Modules/Tooltips/MapIconTooltip.lua [hunk 3]
```diff
@@ -257,6 +260,53 @@ function MapIconTooltip:Show()
                     end
                 end
 
+                if shift and Questie.db.profile.enableTooltipsNextInChain then
+                    -- add quest chain info
+                    local nextQuest = QuestieDB.GetQuest(QuestieDB.GetQuest(questData.questId).nextQuestInChain)
+                    local firstInChain = true;
+                    while nextQuest ~= nil do
+
+                        local nextQuestTitleString = "";
```
Comment (BreakBB) [inline Modules/Tooltips/MapIconTooltip.lua:269]:
```suggestion
                        local nextQuestTitleString;
```

**Comment by Cabro** [comment]
image

This needs some adjusting to not show blacklisted quests.

**Pull request merged**